test/functional/ui/cmdline_spec.lua

```diff
@@ -168,7 +168,7 @@ describe('external cmdline', function()
     ]], nil, nil, function()
       eq({{
         content = { { {}, "default" } },
-        firstc = "",
+        firstc = "@",
```

Comment:
I'm not sure this change is good or not. IIRC emtpy firstc should only occur with `input()` so there is no strong reason to change it (and if not, we should consider setting firstc in that other place instead)

Also, in this file a test should be added that `inputlist()` works as expected with external cmdline (firstc is set to `#`, prompt if applicable)

Reply:
Then also, we should consider allowing GUI:s to convert `inputlist()` in to proper dialogs. That would be a separate follow up PR though (so no action needed here)

Reply:
I think we already use `@` for setting the mode for `input()` function in `getcmdline()` and the likes, so while adding `#` which also uses `input()` function, it seemed better to use `@` as `firstc` to distinguish between the two.

Reply:
Not sure i follow, only `input()` is `input()` (well, except for `inputdialog()` for the moment). Number (`inputlist()` ) is a new and different cmdline type, and should definitely be marked with `firstc = '#'` (even if prompt is available, we will need to update the docs). I was speaking if some other cmdline type  also happened to set `firstc = ""` already (maybe some ex mode state), it is probably much rarer and so nicer to clients to change that instead.

Reply:
Oh, I get it now! There is another mode `:insert` with `firstc = NUL`. The line `ccline.cmdfirstc = (firstc == '@' ? 0 : firstc)` makes `cmdfirstc` for both `:insert` and `input()` same, but I guess that's because they are so similar.
I think I understand it now and should revert back to `firstc = ""` for `@` mode.

Reply:
Yes `:append` /`:insert` should maybe be marked with `-`, but overall ex mode commands will not look well with ext-ui so this could be considered in a later PR (where "later" could mean "never", if the intersection of users using ex mode and extended UI is empty, which is not impossible)